signer/src/transaction_signer.rs

```diff
@@ -425,6 +535,125 @@ where
         Ok(())
     }
 
+    fn create_new_state_machine(&mut self) -> Result<SignerStateMachine, error::Error> {
+        let signers: hashbrown::HashMap<u32, _> = self
+            .signer_public_keys
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(id, key)| {
+                id.try_into()
+                    .map(|id| (id, key))
+                    .map_err(|_| error::Error::TypeConversion)
+            })
+            .collect::<Result<_, _>>()?;
+        let key_ids = signers
+            .clone()
+            .into_iter()
+            .map(|(id, key)| (id + 1, key))
+            .collect();
+
+        let public_keys = wsts::state_machine::PublicKeys { signers, key_ids };
+
+        let threshold = self.threshold;
+        let num_parties = self
+            .signer_public_keys
+            .len()
+            .try_into()
+            .map_err(|_| error::Error::TypeConversion)?;
+        let num_keys = num_parties;
+
+        let signer_pub_key = self.signer_pub_key()?;
+
+        let id: u32 = self
+            .signer_public_keys
+            .iter()
+            .enumerate()
+            .find(|(_, key)| *key == &signer_pub_key)
+            .ok_or_else(|| error::Error::MissingPublicKey)?
+            .0
+            .try_into()
+            .map_err(|_| error::Error::TypeConversion)?;
+
+        let key_ids = vec![id + 1];
+
+        let state_machine = SignerStateMachine::new(
+            threshold,
+            num_parties,
+            num_keys,
+            id,
+            key_ids,
+            self.signer_private_key,
+            public_keys,
+        );
+
+        Ok(state_machine)
+    }
+
+    async fn creat_state_machine_with_loaded_state(
+        &mut self,
+        aggregate_key: &p256k1::point::Point,
+    ) -> Result<SignerStateMachine, error::Error> {
+        let encrypted_shares = self
+            .storage
+            .get_encrypted_dkg_shares(&aggregate_key.x().to_bytes().to_vec())
+            .await?
+            .ok_or(error::Error::MissingDkgShares)?;
+
+        let decrypted = wsts::util::decrypt(
+            &self.signer_private_key.to_bytes(),
+            &encrypted_shares.encrypted_shares,
+        )
+        .map_err(|_| error::Error::Encryption)?;
+
+        let saved_state =
+            wsts::traits::SignerState::decode(decrypted.as_slice()).map_err(error::Error::Codec)?;
+
+        let signer = wsts::v2::Party::load(&saved_state);
```

Comment:
I would add a comment about how `wsts::v2::Party::load` panics if we have more than one party but explain why we do not have to worry about it.

Reply:
Sure, makes sense. I've added a comment about it.